front_end/src/app/(main)/questions/[id]/components/forecast_maker/forecast_maker_conditional/forecast_maker_conditional_binary.tsx

```diff
@@ -241,6 +261,31 @@ const ForecastMakerConditionalBinary: FC<Props> = ({
     }
   };
 
+  const handlePredictWithdraw = async () => {
+    setSubmitErrors([]);
+
+    if (!prevYesForecastValue && !prevNoForecastValue) return;
+
+    const response = await withdrawForecasts(postId, [
+      ...(prevYesForecastValue ? [{ question: questionYesId }] : []),
+      ...(prevNoForecastValue ? [{ question: questionNoId }] : []),
+    ]);
+    questionOptions.forEach((q) => {
+      q.isDirty = false;
+    });
+
+    if (
+      response &&
+      "non_field_errors" in response &&
+      !!response.non_field_errors
+    ) {
```

Comment:
Q: Why do we have a different logic for handling errors when executing withdraw action compared to predict action? These actions seem to return the same payload in catch block.

Reply:
Totally fine point. I don't remember why but I mutated it at some point and then never returned it back. I think it was since I was fiddling with the different error types on the backend and how they effect returned error statements.
Reverting to the predict event format.